gcloud/datastore/_implicit_environ.py

```diff
@@ -126,7 +122,8 @@ def get_connection():
     :rtype: :class:`gcloud.datastore.connection.Connection`
     :returns: A connection defined with the proper credentials.
     """
-    return get_scoped_connection(Connection, SCOPE)
+    credentials = get_credentials()
```

Comment:
I.e., have it take an optional `scope` argument, passed by the subclasses?  We could let the developer pass in `scope` to the subclasses, too, defaulting to the module global.  If they don't want to replace scopes (because `credentials` already has the correct scopes?  How will that be possible, BTW?), then they could pass `scope=None` to the subclass.